plugins/hydra_submitit_launcher/hydra_plugins/hydra_submitit_launcher/config.py

```diff
@@ -1,71 +1,17 @@
 # Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
 from dataclasses import dataclass
-from enum import Enum
 from typing import Optional
 
 from hydra.core.config_store import ConfigStore
 from hydra.types import ObjectConf
 
 
-class QueueType(Enum):
-    auto = "auto"
-    local = "local"
-    slurm = "slurm"
-
-
 @dataclass
-class SlurmQueueConf:
-    # Params are used to configure sbatch, for more info check:
-    # https://github.com/facebookincubator/submitit/blob/master/submitit/slurm/slurm.py
-
-    # maximum time for the job in minutes
-    time: int = 60
-    # number of cpus to use for each task
-    cpus_per_task: int = 10
-    # number of gpus to use on each node
-    gpus_per_node: int = 1
-    # number of tasks to spawn on each node
-    ntasks_per_node: int = 1
-    # number of nodes to use for the job
-    nodes: int = 1
-    # memory to reserve for the job on each node, in GB
-    mem: str = "${hydra.launcher.mem_limit}GB"
-    # slurm partition to use on the cluster
-    partition: Optional[str] = None
-    # USR1 signal delay before timeout
-    signal_delay_s: int = 120
-    # name of the job
-    job_name: str = "${hydra.job.name}"
-    # Maximum number of retries on job timeout.
-    # Change this only after you confirmed your code can handle re-submission
-    # by properly resuming from the latest stored checkpoint.
-    # check the following for more info on slurm_max_num_timeout
-    # https://github.com/facebookincubator/submitit/blob/master/docs/checkpointing.md
-    max_num_timeout: int = 0
-
+class BaseParams:
+    """Configuration shared by all executors
+    """
 
-@dataclass
-class LocalQueueConf:
-    # local executor mocks the behavior of slurm locally
-
-    # maximum time for the job in minutes
-    timeout_min: int = 60
-    # number of gpus to use on each node
-    gpus_per_node: int = 1
-    # number of tasks to spawn on each node (only one node available in local executor)
-    tasks_per_node: int = 1
-
-
-@dataclass
-class AutoQueueConf:
-    # auto executor automatically identifies and uses available cluster
-    # Currently this is only slurm, but local executor can be manually forced
-    # instead.
-    # Most parameters are shared between clusters, some can be cluster specific
-
-    # cluster to use (currently either "slurm" or "local" are supported,
-    # None defaults to an available cluster)
-    cluster: str = "slurm"
+    submitit_folder: str = "${hydra.sweep.dir}/.submitit/%j"
```

Comment:
Let's talk about this for a moment:
With Hydra, each job is getting a working directory inside the sweep dir.

If we can keep all the files for the job in that directory it would be really good (users will have an easier time debugging issues for a particular job).
This may take some submitit changes, but is something to consider for the future.

Right now, if a job is failing the user will need to:
1. Look at the log file in the job output directory.
2. Look in the submitit output folder, figure out which files belong to the job that caused the issue somehow and check the output/error files.

If we managed to use the working directory here it would be better.
I suspect currently this is not possible without changes to submitit.

An alternative solution to this problem is to symlink the stdout and stderr from the submitit folder into the job working directory. But this can also be tricky.

This is not a feature request for this PR, but something to think about to improve the user experience.

Reply:
> If we can keep all the files for the job in that directory it would be really good

Indeed (for now this is the same behavior as before, but I agree it is not convenient)

> I suspect currently this is not possible without changes to submitit.

It would actually be possible if not using arrays, but that's not a good option. With arrays on the other hand we can't specify the exact folder of each run individually, it must be deduced from the parameters of the array. Hacking something is somehow possible but would require a lot of (mostly unhelpful) changes in submitit  and would not be robust anyway.

The other option would be that hydra uses the folder provided by submitit (using the job_id instead of 0, 1, 2 etc). But that's probably not something you'd want either, right?

Reply:
The user can customize the sweep subdirectory through Hydra.
So yeah, I can't really use the submitit generated dir.